### PR TITLE
[@types/tabulator-tables] fixed setData for invoking without any parameter as per official docs

### PR DESCRIPTION
**Pull request by md-redwan-hossain**
Fixes https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/69875

### CODE REVIEW

File: types/tabulator-tables/index.d.ts [hunk 1]
```diff
@@ -2862,7 +2862,7 @@ declare class Tabulator {
     /** Destructor. */
     destroy: () => void;
 
-    setData: (data: any, params?: any, config?: any) => Promise<void>;
+    setData: (data?: any, params?: any, config?: any) => Promise<void>;
```
Comment (mike-lischke) [inline types/tabulator-tables/index.d.ts:2865]: Why's that necessary? `any` includes `undefined`.

Comment (mike-lischke) [reply types/tabulator-tables/index.d.ts:2865]: Turns out `any` (or `unknown` for that matter) is not the same as an optional parameter.

**Review by mike-lischke — APPROVED** [review]

**Comment by mike-lischke** [comment]
@md-redwan-hossain can you also add a simple test case for that change (test with and w/o data)?

File: types/tabulator-tables/index.d.ts (hunk 1 above)
Comment (md-redwan-hossain) [reply types/tabulator-tables/index.d.ts:2865]: I didn't add `any`, it was there before my PR. I just added `?` to make the first parameter optional. However, `?` will make the parameter optional and nullish, it has no relation with the data type.

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/tabulator-tables/tabulator-tables-tests.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 623d399184b5e7f9d631d27532bbaa80798fc336
</p>

**Comment by md-redwan-hossain** [comment]
> @md-redwan-hossain can you also add a simple test case for that change (test with and w/o data)?

Test is added.

**Review by mike-lischke — APPROVED** [review]

**Comment by mike-lischke** [comment]
Thanks @md-redwan-hossain

**Comment by md-redwan-hossain** [comment]
Ready to merge

**Pull request merged**